### ISSUE TITLE
Coral.ai USB not being found within container.  

### ISSUE THREAD

**Issue by grasponcrypto** [blakeblackshear/frigate#1584]
I'll preface this with its likely a user error.  However, nothing i have tried has worked, and from documentation I know of nothing else to try, so im a bit stuck.  I used frigate with cpu as a test and it worked great.  Got the coral usb and I simply cannot get frigate to find it.  Hoping someone has experienced this and has some tips to try.

Things I have already tried:
unplug and plug it (coral usb) back in
reboot host
change usb ports
using docker
using podman
using podman-compose
removing detector portion of config - still fails to find it, but doesnt keep restarting the container

ubuntu 20.04 x64 bit
podman (tried docker as well)

```
lsusb -v

Bus 002 Device 002: ID 1a6e:089a Global Unichip Corp. 
```

```
podman exec -t frigate find /dev/bus
/dev/bus
/dev/bus/usb
/dev/bus/usb/001
/dev/bus/usb/001/001
/dev/bus/usb/001/002
/dev/bus/usb/001/003
/dev/bus/usb/002
/dev/bus/usb/002/001
/dev/bus/usb/002/002
```

config.yml:
```
mqtt:
  host: 192.168.1.10
  port: 1883
  user: frigate
  password: passphrase
cameras:
  front_door:
    ffmpeg:
      inputs:
        - path: rtsp://rspuser:{FRIGATE_RTSP_PASSWORD}@192.168.1.210:8554/unicast
          roles:
            - clips
            - detect
            - rtmp
    width: 1280
    height: 720
    fps: 5
    zones:
      zone_0:
        coordinates: 618,643,621,189,443,202,441,599
      zone_1:
        coordinates: 1241,720,1255,167,895,172,868,626
      zone_2:
        coordinates: 457,254,373,252,355,452,452,498
clips:
  retain:
    default: 20
    objects:
      person: 45
detectors:
  coral:
    type: edgetpu
    device: usb
  cpu:
    type: cpu
```

docker-compose.yml
```
version: "3.9"
services:
  frigate:
    container_name: frigate
    restart: unless-stopped
    image: blakeblackshear/frigate:stable-amd64
    devices:
       - /dev/bus/usb/:/dev/bus/usb/
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /home/frigate/config.yml:/config/config.yml:ro
      - /sanman/Podman/cameras:/media/frigate
      - type: tmpfs
        target: /tmp/cache
        tmpfs:
          size: 1000000000
    ports:
      - "5000:5000"
      - "1935:1935"
    environment:
      FRIGATE_RTSP_PASSWORD: "12345"
```

```
podman logs frigate
 * Starting nginx nginx
   ...done.
frigate.app                    WARNING : Camera front_door has clips assigned to an input, but clips is not enabled.
Starting migrations
peewee_migrate                 INFO    : Starting migrations
There is nothing to migrate
peewee_migrate                 INFO    : There is nothing to migrate
frigate.mqtt                   INFO    : MQTT connected
detector.coral                 INFO    : Starting detection process: 35
cameras@ajalon-Rampage-Formula:~$ podman logs frigate
 * Starting nginx nginx
   ...done.
frigate.app                    WARNING : Camera front_door has clips assigned to an input, but clips is not enabled.
Starting migrations
peewee_migrate                 INFO    : Starting migrations
There is nothing to migrate
peewee_migrate                 INFO    : There is nothing to migrate
frigate.mqtt                   INFO    : MQTT connected
detector.coral                 INFO    : Starting detection process: 35
detector.cpu                   INFO    : Starting detection process: 37
frigate.app                    INFO    : Camera processor started for front_door: 40
frigate.edgetpu                INFO    : Attempting to load TPU as usb
Process detector:coral:
frigate.edgetpu                INFO    : No EdgeTPU detected.
frigate.app                    INFO    : Capture process started for front_door: 41
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/tflite_runtime/interpreter.py", line 152, in load_delegate
    delegate = Delegate(library, options)
  File "/usr/local/lib/python3.8/dist-packages/tflite_runtime/interpreter.py", line 111, in __init__
    raise ValueError(capture.message)
ValueError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/frigate/frigate/edgetpu.py", line 124, in run_detector
    object_detector = LocalObjectDetector(tf_device=tf_device, num_threads=num_threads)
  File "/opt/frigate/frigate/edgetpu.py", line 63, in __init__
    edge_tpu_delegate = load_delegate('libedgetpu.so.1.0', device_config)
  File "/usr/local/lib/python3.8/dist-packages/tflite_runtime/interpreter.py", line 154, in load_delegate
    raise ValueError('Failed to load delegate from {}\n{}'.format(
ValueError: Failed to load delegate from libedgetpu.so.1.0

frigate.watchdog               INFO    : Detection appears to have stopped. Exiting frigate...
frigate.app                    INFO    : Stopping...
```

**Comment by DeviantUK** [comment]
Have you tried `privileged: true` from the docker-compose documentation at https://blakeblackshear.github.io/frigate/installation ?

**Comment by grasponcrypto** [comment]
yes, that was in the docker-compose file until the last few attempts.  Did not make a difference.  I also tried, just now, doing so using the podman run command.  Still no such luck.  Thank you for the suggestion though!

**Comment by blakeblackshear** [comment]
Try moving the usb section to a volume instead of a device.

**Comment by grasponcrypto** [comment]
So, I think it may be a podman rootless issue.  Not positive there, but using this docker run command it is finally working.
```
  docker run -d --name frigate --restart=unless-stopped --mount type=tmpfs,target=/tmp/cache,tmpfs-size=1000000000 -v /dev/bus/usb:/dev/bus/usb --privileged -v /sanman/cameras:/media/frigate -v /home/cameras/config.yml:/config/config.yml:ro -v /etc/localtime:/etc/localtime:ro -e FRIGATE_RTSP_PASSWORD='12345' -p 5000:5000 -p 1935:1935 blakeblackshear/frigate:stable-amd64
```

however, with podman using the same exact command, except substituting docker with podman, it fails.  I imagine its a permissions issue.  Does anyone know if frigate works as expected using docker rootless?

Anyway, it is now working.  Thank you both!

**Comment by NYKoop** [comment]
I know this is an old post, but I was in the same situation with Frigate not recognizing the Coral USB TPU while running in Podman. I was able to get it to start by adding the —privileged to the volume mount of /dev/bus/usb just as grasponcrypto stated above. The other key thing is doing a volume mount of /dev/bus/usb rather than a device. I know he said he couldn’t get Podman to work with this option a couple years ago, but I’m thinking a bug was fixed since then. I figured I’d drop this message here in case someone else was struggling to get this to work with Podman.
